fix: adapt function signatures for GCC 15 compatibility

hillwoodroc opened the pull request:
GCC 15 introduced stricter checks on function declarations and definitions, requiring consistent signatures across calls and headers. This patch updates several function declarations to include explicit parameter types where they were previously omitted, which caused compilation errors under GCC 15.

Log: Changes include:
- `start_encoder_thread()` updated to `start_encoder_thread(void *)`
- `v4l2core_check_device_list_events()` updated to accept `v4l2_dev_t *vd`
- Adjusted corresponding call sites to pass required arguments

Also fixed a mismatched call to `v4l2core_time_get_timestamp()` which no longer requires parameters.

lzwind commented on src/multimedia/camera/libcam/libcam/camview.h:
```diff
@@ -397,7 +397,7 @@ audio_context_t *get_audio_context(void);
  * returns: error code
  */
-int start_encoder_thread();
+int start_encoder_thread(void *);
```
/home/uos/Desktop/fcitx5-pinyin-zhwiki-master/dtkmultimedia/src/multimedia/camera/dcamera.cpp: In member function ‘void Dtk::Multimedia::DCamera::takeVideo(const QString&)’:
/home/uos/Desktop/fcitx5-pinyin-zhwiki-master/dtkmultimedia/src/multimedia/camera/dcamera.cpp:213:29: error: too few arguments to function ‘int start_encoder_thread(void*)’
  213 |         start_encoder_thread();


**There is an error that needs to be processed**

hillwoodroc replied:
不太理解为何会出错，这个函数的实现的确是以一个指针为形参。
https://github.com/linuxdeepin/dtkmultimedia/blob/master/src/multimedia/camera/libcam/libcam/camview.c#L1291

hillwoodroc replied:
注释里面清楚的写了：
start the encoder thread
args:
    data - pointer to user data

lzwind replied:
注释有问题,建议编译下试试

hillwoodroc replied:
我这儿是能编译的，如果不改这个函数申明，就是会报签名错误。

```
int start_encoder_thread(void *data)
{
    int ret = __THREAD_CREATE(&encoder_thread, encoder_loop, data);

    if(ret)
        fprintf(stderr, "deepin-camera: encoder thread creation failed (%i)\n", ret);
    else if(debug_level > 2)
        printf("deepin-camera: created encoder thread with tid: %u\n",
            (unsigned int) encoder_thread);

    return ret;
}
```
这个函数的确是以一个 void 指针作为参数的。

lzwind replied:
/*
 * start the encoder thread
 * args:
 *   none
 *
 * asserts:
 *   none
 *
 * returns: error code
 */
int start_encoder_thread() {
  int ret = __THREAD_CREATE(&encoder_thread, encoder_loop, NULL);
  .......
}   建议把实现函数同步改下

hillwoodroc replied:
已同步，重新提交。

hillwoodroc committed:
fix: Adapt function signatures for GCC 15 compatibility

GCC 15 introduced stricter checks on function declarations and
definitions, requiring consistent signatures across calls and
headers. This patch updates several function declarations to
include explicit parameter types where they were previously
omitted, which caused compilation errors under GCC 15.

Log:

lzwind commented:
/merge
